Compile examples on travis when repo is updated

malramsay64 opened the pull request:
Including the examples in the git repository is not a great idea, especially when they are being
updated regularly. This uses travis to check that all the examples are going to compile, and on
tagged releases upload to the github releases.

malramsay64 committed:
Create single examples directory

Put all the examples in a single directory to make it easier to build
all the examples with a single Makefile.

This also removes all the generated pdf files from the repository to
prevent the size ballooning as I continue to make changes. The idea is
to include the generated files as an embed in the README

malramsay64 committed:
Create build script for travis-ci

malramsay64 committed:
Install texlive-full on travis

malramsay64 committed:
Include references for poster example

malramsay64 committed:
Ignore all pdf files

malramsay64 committed:
Run script which will pass to build cache

malramsay64 committed:
Fix bug in poster references

malramsay64 committed:
Reduce output from latexmk

malramsay64 committed:
Use miktex as latex engine

malramsay64 committed:
Specify conda-forge repository properly

malramsay64 committed:
Build with miktex docker container

malramsay64 committed:
Reconfigure travis to use docker

malramsay64 committed:
Ensure $HOME/.miktex is created on travis

malramsay64 committed:
Change directory to save miktex files

malramsay64 committed:
More output

malramsay64 committed:
Configure setup

malramsay64 committed:
Use correct command to update-db

malramsay64 committed:
Update or install latexmk

malramsay64 committed:
Better conditional

malramsay64 committed:
make output dir

malramsay64 committed:
Change permissions of directories

malramsay64 committed:
Update tex flags

malramsay64 committed:
Include pdf version of logo

malramsay64 committed:
Install biber using apt

malramsay64 committed:
Don't install recommends

malramsay64 committed:
remove biber requriement

malramsay64 committed:
apparently biber is default

malramsay64 committed:
Use proper include paths

malramsay64 committed:
Build with tectonic and an installed biber

malramsay64 committed:
Update travis cache and installation

malramsay64 committed:
Yes I want to install tectonic

malramsay64 committed:
Get paths correct

jnothman commented:
(It's very comforting to know there are academics here who know how to do
this stuff.)​

malramsay64 committed:
Get newlines correct

malramsay64 committed:
Check for correct file

malramsay64 committed:
Fix typo

malramsay64 committed:
Create output directory

malramsay64 committed:
Use bash instead of make conditional

malramsay64 committed:
usyd-beamer-theme 0.1.0-dev0

- Test deployment of travis

malramsay64 committed:
Use correct path for bcf file

malramsay64 committed:
Use correct travis api key

malramsay64 committed:
usyd-beamer-theme 0.1.0-dev1

- Update deployment configuration

malramsay64 committed:
Add logobar examle to examples

malramsay64 committed:
Correct output directory of biber

malramsay64 committed:
Ensure output bcf file exists

malramsay64 committed:
Create bbl file in current dir

malramsay64 committed:
usyd-beamer-theme 0.1.0-dev2

- Add logobar examle to examples
- Configure biber correctly

malramsay64 committed:
Bugfix - get make clean to remove all generated files

malramsay64 committed:
Include links to release versions in README

Now I am building the releases rather than including them in the
repository I have included links in the README to the respective files
which are representative of each theme element.

malramsay64 commented:
A [writeup](https://malramsay.com/post/compiling_latex_on_travis/) summarising the results of this process.
